src/xmipp/libraries/reconstruction/flexible_alignment.cpp

```diff
@@ -37,1409 +39,822 @@
-                  free(DRz1);
-                  free(DRy);
-                  free(DRz2);
-                  free(DR0);
-                  free(DR1);
-                  free(DR2);
-                  free(Tr);
-                  return(ERROR);
-              }
-
-
-              for( i=0;i<trialSize;i++)
-              {
-                  Gradient[i] = 0.0;
-                  for( j=0;j<trialSize;j++)
-                  {
-                      Hessian[i*trialSize+j] = 0.0;
-                  }
-              }
-              std::cout << " 1352 phi = " << phi << std::endl;
-              for(int kx=0;kx<Xwidth;kx++)
-                  for(int ky=0;ky<Ywidth;ky++)
-                  {
-                      difference = rg_projimage(kx-half_Xwidth,ky-half_Ywidth) - P_esp_image(kx-half_Xwidth,ky-half_Ywidth);
-                      //std::cout << " 1399 difference = " << difference << std::endl;
-                      helpgr[0] = DP_Rx(kx,ky); //std::cout << "  helpgr= " << helpgr[0];
-                      helpgr[1] = DP_Ry(kx,ky); //std::cout << "||" << helpgr[1];
-                      helpgr[2] = DP_Rz2(kx,ky);//std::cout << "||" << helpgr[2];
-                      helpgr[3] = DP_x(kx,ky);  //std::cout << "||" << helpgr[3];
-                      helpgr[4] = DP_y(kx,ky);  //std::cout << "||" << helpgr[4];
-                      for(j=0;j<dim;j++)
-                      {
-                          helpgr[5+j] =DP_q(j,kx,ky);//std::cout << "||" << helpgr[5+j];
-                      }
-                      gradhesscost_atpixel(Gradient,Hessian,helpgr,difference);
-                      /*if (gradhesscost_atpixel(Gradient,Hessian,helpgr,difference) == ERROR)
-                  {
-                       WRITE_ERROR(return_gradhesscost, "Error returned by partialpfunction");
-                       //free(defamp);
-                       free(DP_q);
-                       free(DP_y);
-                       free(DP_x);
-                       free(DP_Rz2);
-                       free(DP_Ry);
-                       free(DP_Rz1);
-                       free(Rz1);
-                       free(Ry);
-                       free(Rz2);
-                       free(R);
-                       free(DRz1);
-                       free(DRy);
-                       free(DRz2);
-                       free(DR0);
-                       free(DR1);
-                       free(DR2);
-                       free(Tr);
-                       return(ERROR);
-                       }*/
-                  }
-              for (i=0;i<trialSize;i++)
-              {
-                  for(j=0;j<trialSize;j++)
-                      std::cout << Hessian[i*trialSize+j] << "||";
-                  std::cout << std::endl;
-              }
-
-              std::cout << " 1392 phi = " << phi << std::endl;
-
-              for (int i=0;i<trialSize;i++)
-              {
-                  Hessian[i*trialSize + i] += lambda*Hessian[i*trialSize + i] ;
```

Comment:
all this code probably should not be changed

Reply:
Sonarcloud was giving many errors related to memory management. There are two issues here: 1) Sonarcloud promotes that there is automatic memory management; 2) There were many objects that, depending on the path, were allocated, but not released.